app/src/main/java/org/oppia/android/app/options/AppLanguageFragment.kt

```diff
@@ -7,29 +7,44 @@ import android.view.View
 import android.view.ViewGroup
 import org.oppia.android.app.fragment.FragmentComponentImpl
 import org.oppia.android.app.fragment.InjectableFragment
-import org.oppia.android.util.extensions.getStringFromBundle
+import org.oppia.android.app.model.AppLanguageFragmentArguments
+import org.oppia.android.app.model.AppLanguageFragmentStateBundle
+import org.oppia.android.app.model.OppiaLanguage
+import org.oppia.android.util.extensions.getProto
+import org.oppia.android.util.extensions.putProto
 import javax.inject.Inject
 
-private const val APP_LANGUAGE_PREFERENCE_TITLE_ARGUMENT_KEY =
-  "AppLanguageFragment.app_language_preference_title"
-private const val APP_LANGUAGE_PREFERENCE_SUMMARY_VALUE_ARGUMENT_KEY =
-  "AppLanguageFragment.app_language_preference_summary_value"
-private const val SELECTED_LANGUAGE_SAVED_KEY = "AppLanguageFragment.selected_language"
-
 /** The fragment to change the language of the app. */
 class AppLanguageFragment : InjectableFragment(), AppLanguageRadioButtonListener {
 
   @Inject
   lateinit var appLanguageFragmentPresenter: AppLanguageFragmentPresenter
 
   companion object {
-    fun newInstance(prefsKey: String, prefsSummaryValue: String): AppLanguageFragment {
-      val fragment = AppLanguageFragment()
-      val args = Bundle()
-      args.putString(APP_LANGUAGE_PREFERENCE_TITLE_ARGUMENT_KEY, prefsKey)
-      args.putString(APP_LANGUAGE_PREFERENCE_SUMMARY_VALUE_ARGUMENT_KEY, prefsSummaryValue)
-      fragment.arguments = args
-      return fragment
+    private const val FRAGMENT_ARGUMENTS_KEY = "AppLanguageFragment.arguments"
+    private const val FRAGMENT_SAVED_STATE_KEY = "AppLanguageFragment.saved_state"
+
+    fun newInstance(prefsSummaryValue: OppiaLanguage): AppLanguageFragment {
```

Comment:
For this & all other files modified in this PR, please make sure they're all removed from the kdoc validity exemptions textproto & then update/add KDocs as needed per your changes.

Reply:
Thanks updated, the same.

Reply:
This doesn't seem resolved yet--new KDocs will need to be added for methods like this that are currently missing them (which is why I suggested to remove all files you've changed from the KDocs exemption list so that you know which will require updating).

Reply:
Still unresolved @KevinGitonga, please fix.

Reply:
Thanks @BenHenning updated kdocs for most files on this, most of the removals would include the static profileId declarations used through out was thinking of removing the remaining while on profile_id at #4869 since i have removed most instances of the values.